### PR TITLE
Make it so the assignment time column filter is dynamic and api driven

### PR DESCRIPTION
**Pull request by reillz10**

### CODE REVIEW

File: src/components/Gradebook/index.jsx [hunk 1]
```diff
@@ -260,6 +283,16 @@ export default class Gradebook extends React.Component {
                     </label>
                   </span>
                 </div>
+                { this.props.assignmnetTypes.length > 0 &&
```
Comment (schenedx) [inline src/components/Gradebook/index.jsx]: With this drop down, can you please remove the `category` radio button elements on top?

File: src/components/Gradebook/index.jsx [hunk 2]
```diff
@@ -260,6 +283,16 @@ export default class Gradebook extends React.Component {
...
+                  <div className="student-filters">
+                    <InputSelect
```
Comment (schenedx) [inline src/components/Gradebook/index.jsx]: Can you provide a label to this drop down? I think you can just call it `Assignment Types`, just like how Studio does it.

File: src/components/Gradebook/index.jsx [hunk 3]
```diff
@@ -74,6 +75,15 @@ export default class Gradebook extends React.Component {
     return `?${queryString.stringify(parsed)}`;
   };
 
+  mapAssignmentTypeEntries = (entries) => {
+    const mapped = entries.map(entry => ({
+      id: entry,
+      label: entry,
+    }));
+    mapped.unshift({ id: 0, label: 'All-Assignments' });
```
Comment (schenedx) [inline src/components/Gradebook/index.jsx]: If you had the label besides the overall dropdown, you can have this entry to be `All`. That should be clear enough

File: src/components/Gradebook/index.jsx [hunk 4]
```diff
@@ -122,6 +136,15 @@ export default class Gradebook extends React.Component {
     this.props.history.push(updatedQueryStrings);
   };
 
+  mapSelectedAssignmentTypeEntry = (entry) => {
+    const selectedAssignmentTypeEntry = this.props.assignmentTypes
+      .find(x => x.id === parseInt(entry, 10));
+    if (selectedAssignmentTypeEntry) {
+      return selectedAssignmentTypeEntry.name;
+    }
+    return 'All-Assignments';
```
Comment (schenedx) [inline src/components/Gradebook/index.jsx]: Again, I think we should just change this label to `All`

File: src/data/actions/grades.js [hunk 5]
```diff
@@ -72,7 +73,7 @@ const fetchGrades = (courseId, cohort, track, showSuccess) => (
           data.results.sort(sortAlphaAsc),
           cohort,
           track,
-          headingMapper.all(dispatch, data.results[0]),
+          headingMapper('All-Assignments')(dispatch, data.results[0]),
```
Comment (schenedx) [inline src/data/actions/grades.js]: Maybe move this label text to a single const variable? Please update this to `All`

File: src/data/apiClient.js [hunk 6]
```diff
@@ -10,6 +10,7 @@ const apiClient = getAuthenticatedAPIClient({
   refreshAccessTokenEndpoint: configuration.REFRESH_ACCESS_TOKEN_ENDPOINT,
   accessTokenCookieName: configuration.ACCESS_TOKEN_COOKIE_NAME,
   csrfCookieName: configuration.CSRF_COOKIE_NAME,
+  studioBaseUrl: configuration.STUDIO_BASE_URL,
```
Comment (schenedx) [inline src/data/apiClient.js]: This is no longer needed. Please remove.

File: src/components/Gradebook/index.jsx (hunk 1 above)
Comment (reillz10) [reply src/components/Gradebook/index.jsx]: yes! done!

File: src/components/Gradebook/index.jsx (hunk 2 above)
Comment (reillz10) [reply src/components/Gradebook/index.jsx]: 👍

**Pull request merged**